### PR TITLE
Ordered trie

### PR DESCRIPTION
**Pull request by arnetheduck**
Speed up trie computations and remove redundant ways of performing this operation.

### CODE REVIEW

File: fluffy/network/history/history_network.nim [hunk 1]
```diff
@@ -11,12 +11,12 @@ import
...
   chronos,
   chronicles,
-  eth/[trie, trie/db],
+  eth/trie/ordered_trie,
   eth/common/[hashes, headers_rlp, blocks_rlp, receipts_rlp, transactions_rlp],
   eth/p2p/discoveryv5/[protocol, enr],
   ../../common/common_types,
   ../../database/content_db,
-  ../../network_metadata,
+  ../../network_me tadata,
```
Comment (kdeme) [inline fluffy/network/history/history_network.nim:19]: This is strange, I'm surprised most of CI passes?

Comment (arnetheduck) [reply fluffy/network/history/history_network.nim]: `conten t_db` doesn't work but this very clearly does .. weird bug.

**Comment by jangko** [comment]
failing testutp when bumping nimbus-eth2:
https://github.com/status-im/nimbus-eth1/actions/runs/11248428316

**Comment by kdeme** [comment]
> failing testutp when bumping nimbus-eth2: https://github.com/status-im/nimbus-eth1/actions/runs/11248428316

Yes, I also saw this when reproducing locally, by executing the same steps as in the Docker file.

```
ERROR: Nim package already present in '/home/deme/repos/nimbus-eth1-test/vendor/.nimble/pkgs/nim-minilru-#head/nim-minilru.nimble-link': '/home/deme/repos/nimbus-eth1-test/vendor/nim-minilru/src'
Will not replace it with '/home/deme/repos/nimbus-eth1-test/vendor/nimbus-eth2/vendor/nim-minilru/src'.
Pick one and put the other's relative path in EXCLUDED_NIM_PACKAGES.
See also: https://github.com/status-im/nimbus-build-system#excluded_nim_packages

fatal: run_command returned non-zero status for vendor/nimbus-eth2/vendor/nim-minilru
.
fatal: run_command returned non-zero status while recursing in the nested submodules of vendor/nimbus-eth2
.
make[1]: *** [vendor/nimbus-build-system/makefiles/targets.mk:152: vendor/.nimble] Error 128
make[1]: *** Waiting for unfinished jobs....
****
```

**Comment by arnetheduck** [comment]
yay, green - thanks!

**Pull request merged**